core/services/log/orm.go

```diff
@@ -152,7 +163,18 @@ ON CONFLICT (%[1]s, block_hash, log_index) WHERE %[1]s IS NOT NULL DO UPDATE SET
 	}
 
 	stmt := fmt.Sprintf(q, jobIDName)
-	return o.db.Exec(stmt, args...).Error
+	err = o.db.Exec(stmt, args...).Error
+	switch v := err.(type) {
+	case *pgconn.PgError:
+		if v.Code == "23503" && v.ConstraintName == "log_broadcasts_eth_log_id_fkey" {
+			// NOTE: This rare case indicates that the eth_log was deleted
```

Comment:
Should we log here? (and above?) Maybe a warning?

Reply:
I'm not really sure it indicates any type of error, I think it's expected occasionally in normal operation.

Reply:
@spooktheducks can you confirm?

Reply:
I spoke with spook, this can legitimately happen but I pushed debug logging anyway